File: lib/services/token_service.dart
import 'package:shared_preferences/shared_preferences.dart';

class TokenService {
  static const String _tokenKey = 'auth_token';
  static const String _userIdKey = 'user_id';
  static const String _userEmailKey = 'user_email';
  static const String _userNameKey = 'user_name';

  /// Save JWT token to local storage
  static Future<bool> saveToken(String token) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      return await prefs.setString(_tokenKey, token);
    } catch (e) {
      print('Error saving token: $e');
      return false;
    }
  }

  /// Get JWT token from local storage
  static Future<String?> getToken() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      return prefs.getString(_tokenKey);
    } catch (e) {
      print('Error retrieving token: $e');
      return null;
    }
  }

  /// Check if token exists
  static Future<bool> hasToken() async {
    try {
      final token = await getToken();
      return token != null && token.isNotEmpty;
    } catch (e) {
      print('Error checking token: $e');
      return false;
    }
  }

  /// Save user information
  static Future<bool> saveUserInfo({
    required int userId,
    required String email,
    required String firstName,
    required String lastName,
  }) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      await prefs.setInt(_userIdKey, userId);
      await prefs.setString(_userEmailKey, email);
      await prefs.setString(_userNameKey, '$firstName $lastName');
      return true;
    } catch (e) {
      print('Error saving user info: $e');
      return false;
    }
  }

  /// Get user ID
  static Future<int?> getUserId() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      return prefs.getInt(_userIdKey);
    } catch (e) {
      print('Error retrieving user ID: $e');
      return null;
    }
  }

  /// Get user email
  static Future<String?> getUserEmail() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      return prefs.getString(_userEmailKey);
    } catch (e) {
      print('Error retrieving user email: $e');
      return null;
    }
  }

  /// Get user name
  static Future<String?> getUserName() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      return prefs.getString(_userNameKey);
    } catch (e) {
      print('Error retrieving user name: $e');
      return null;
    }
  }

  /// Clear all authentication data (logout)
  static Future<bool> clearAll() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      await prefs.remove(_tokenKey);
      await prefs.remove(_userIdKey);
      await prefs.remove(_userEmailKey);
      await prefs.remove(_userNameKey);
      return true;
    } catch (e) {
      print('Error clearing authentication data: $e');
      return false;
    }
  }
}

File: react_frontend_web_backup/src/services/tokenService.ts
const TOKEN_KEY = 'auth_token';
const USER_ID_KEY = 'user_id';
const USER_EMAIL_KEY = 'user_email';
const USER_NAME_KEY = 'user_name';

export const tokenService = {
  saveToken: (token: string): boolean => {
    try {
      localStorage.setItem(TOKEN_KEY, token);
      return true;
    } catch (e) {
      console.error('Error saving token:', e);
      return false;
    }
  },

  getToken: (): string | null => {
    try {
      return localStorage.getItem(TOKEN_KEY);
    } catch (e) {
      console.error('Error retrieving token:', e);
      return null;
    }
  },

  hasToken: (): boolean => {
    const token = tokenService.getToken();
    return !!token && token.length > 0;
  },

  saveUserInfo: (userId: number, email: string, firstName: string, lastName: string): boolean => {
    try {
      localStorage.setItem(USER_ID_KEY, userId.toString());
      localStorage.setItem(USER_EMAIL_KEY, email);
      localStorage.setItem(USER_NAME_KEY, `${firstName} ${lastName}`);
      return true;
    } catch (e) {
      console.error('Error saving user info:', e);
      return false;
    }
  },

  getUserId: (): number | null => {
    try {
      const id = localStorage.getItem(USER_ID_KEY);
      return id ? parseInt(id, 10) : null;
    } catch (e) {
      console.error('Error retrieving user ID:', e);
      return null;
    }
  },

  getUserEmail: (): string | null => {
    try {
      return localStorage.getItem(USER_EMAIL_KEY);
    } catch (e) {
      console.error('Error retrieving user email:', e);
      return null;
    }
  },

  getUserName: (): string | null => {
    try {
      return localStorage.getItem(USER_NAME_KEY);
    } catch (e) {
      console.error('Error retrieving user name:', e);
      return null;
    }
  },

  clearAll: (): boolean => {
    try {
      localStorage.removeItem(TOKEN_KEY);
      localStorage.removeItem(USER_ID_KEY);
      localStorage.removeItem(USER_EMAIL_KEY);
      localStorage.removeItem(USER_NAME_KEY);
      return true;
    } catch (e) {
      console.error('Error clearing authentication data:', e);
      return false;
    }
  }
};
